gtsam/nonlinear/PriorFactor.h

```diff
@@ -91,7 +95,7 @@ namespace gtsam {
     /** implement functions needed to derive from Factor */
 
     /** vector of errors */
-    Vector evaluateError(const T& x, boost::optional<Matrix&> H = boost::none) const override {
+    Vector evaluateError(const T& x, OptionalMatrixType H) const override {
```

Comment:
Just for reference, is it fair to say that this (and a lot of other instances) could be migrated by doing global find-replace: `boost::optional<Matrix&>` -> `OptionalMatrixType` ?

Reply:
Yea that's right as long as the the operator overloads for `*` and `->` have the same semantics, if I got that right